src/QMCWaveFunctions/SPOSet.h

```diff
@@ -157,6 +157,21 @@ class SPOSet : public QMCTraits
   virtual void checkInVariables(opt_variables_type& active) {}
   virtual void checkOutVariables(const opt_variables_type& active) {}
 
+  virtual void evaluateDerivatives(ParticleSet& P,
+                                          const opt_variables_type& optvars,
+                                          std::vector<ValueType>& dlogpsi,
+                                          std::vector<ValueType>& dhpsioverpsi,
+                                          const ValueType& psiCurrent,
+                                          const ValueMatrix_t& M,
+                                          const ValueMatrix_t& Minv,
+                                          const ValueMatrix_t& Bbar,
+                                          const int m_first_var_pos,
+                                          const std::vector<std::pair<int, int>> m_act_rot_inds,
+                                          const int m_nel,
```

Comment:
All the rotation related parameters need to be encapsulated inside the SPO class. The determinant class doesn't need to know any details how SPO is optimized. Thus, m_act_rot_inds probably should not show up in the arguments.

What is Bbar? Please document the added function as well all the arguments. Please do the same thing for evaluateDerivatives used by MSD as well.